k8s/k8s.go

```diff
@@ -91,19 +93,47 @@ func (k8s *Client) PodByIP(IP string) (*v1.Pod, error) {
 	}
 
 	if len(pods) == 0 {
-		return nil, fmt.Errorf("pod with specified IP not found")
+		metrics.PodNotFoundInCache.Inc()
+		return nil, fmt.Errorf("pod with specificed IP not found")
 	}
 
 	if len(pods) == 1 {
 		return pods[0].(*v1.Pod), nil
 	}
 
-	//This happens with `hostNetwork: true` pods
-	podNames := make([]string, len(pods))
-	for i, pod := range pods {
-		podNames[i] = pod.(*v1.Pod).ObjectMeta.Name
+	if !k8s.dealWithDupIP {
+		podNames := make([]string, len(pods))
+		for i, pod := range pods {
+			podNames[i] = pod.(*v1.Pod).ObjectMeta.Name
+		}
+		return nil, fmt.Errorf("%d pods (%v) with the ip %s indexed", len(pods), podNames, IP)
+	}
+	pod, err := dealWithDuplicatedIP(k8s, IP)
```

Comment:
```suggestion
	pod, err := resolveDuplicatedIP(k8s, IP)
```

Consistency with other suggestions.